starter_templates/haskell/app/Main.hs

```diff
@@ -1,31 +1,20 @@
+{-# OPTIONS_GHC -Wno-unused-top-binds #-}
+{-# LANGUAGE BlockArguments #-}
 {-# LANGUAGE OverloadedStrings #-}
-module Main where
 
+module Main (main) where
 
-import Network.Socket
-    ( SocketType(..)
-    , Family(..)
-    , SocketOption(..)
-    , SockAddr(..)
-    , socket
-    , listen
-    , bind
-    , accept
-    , setSocketOption
-    , defaultProtocol
-    )
-
+import Network.Simple.TCP (serve, HostPreference(HostAny), closeSock)
 
 
 main :: IO ()
 main = do
     -- You can use print statements as follows for debugging, they'll be visible when running tests.
     putStrLn "Logs from your program will appear here"
 
-    -- Uncomment this to pass stage 1
-    -- sock <- socket AF_INET Stream defaultProtocol
-    -- setSocketOption sock ReuseAddr 1
-    -- bind sock (SockAddrInet 6379 0)
-    -- listen sock 5
-    -- _ <- accept sock
-    -- return ()
+    -- Uncomment this block to pass stage 1
+    -- let port = "6379"
+    -- putStrLn $ "\r\n>>> Redis server listening on port " ++ port ++ " <<<"
+    -- serve HostAny port $ \(socket, _address) -> do
```

Comment:
Yes, it is a convention, but also somewhat a necessity for unused variables. Since the `address` return value is not used, one could also replace it with `_`, especially to avoid GHC warning you that you have unused variables. However, by using `_address` you know which value is returned, but also GHC seems to be happy with that :)
See https://stackoverflow.com/questions/50705757/what-in-haskell-does-an-underscore-preceding-a-variable-mean

Reply:
I just realized that _address is used as part of the 'logging', i.e. `show _address`, I'll change that to `show address` where necessary to adhere to this logic.